client/src/test/java/io/pravega/client/segment/impl/SegmentOutputStreamTest.java

```diff
@@ -657,4 +663,41 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
         assertEquals(false, ack.isDone());
     }
     
+    @Test(timeout = 10000)
+    public void testNoSuchSegment() throws Exception {
+        UUID cid = UUID.randomUUID();
+        PravegaNodeUri uri = new PravegaNodeUri("endpoint", SERVICE_PORT);
+
+        MockConnectionFactoryImpl cf = new MockConnectionFactoryImpl();
+        ScheduledExecutorService executor = mock(ScheduledExecutorService.class);
+        implementAsDirectExecutor(executor); // Ensure task submitted to executor is run inline.
+        cf.setExecutor(executor);
+
+        MockController controller = new MockController(uri.getEndpoint(), uri.getPort(), cf);
+        ClientConnection connection = mock(ClientConnection.class);
+        cf.provideConnection(uri, connection);
+        SegmentOutputStreamImpl output = new SegmentOutputStreamImpl(SEGMENT, controller, cf, cid, segmentSealedCallback, RETRY_SCHEDULE);
+        output.reconnect();
+        verify(connection).send(new SetupAppend(1, cid, SEGMENT));
+
+        cf.getProcessor(uri).noSuchSegment(new NoSuchSegment(1, SEGMENT)); // simulate a connection dropped
```

Comment:
This comment might be misleading. Calling `noSuchSegment` closes the connection:

https://github.com/pravega/pravega/pull/1811/files#diff-698633adec608f170863e75caed6297dR309

so it is a consequence of invoking `noSuchSegment`.  What this is really doing is simulating that the service is returning no such segment and that as a consequence we are closing the connection.